### PR TITLE
Helper methods for summarizing contributions and CSV export (browser-laptop#3477)

### PR DESCRIPTION
**Pull request by willy-b**
Adds CSV export for ledger transaction/contribution data by publisher
  supports brave/browser-laptop#3477
  needed for PR https://github.com/brave/browser-laptop/pull/4312

A couple contribution summary helper methods:

- Client#_publisherVoteData([viewingIds])
  get a contribution summary organized by publisher for 1, N (an array), or all transactions
  (useful for brave/browser-laptop#3477)

- Client#_totalContribution([viewingIds])
  get the total amount contributed in BTC and fiat for 1, N (an array), or all transactions

- Client#_transactionByViewingId(viewingId) (self-explanatory)

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -275,6 +275,160 @@ Client.prototype.ballots = function (viewingId) {
   return count
 }
 
+Client.prototype._totalContribution = function (viewingIds) {
```
Comment (mrose17) [inline index.js]: what is the entry point that users of this module are going to access? it should be the function that doesn't start with a "_" and is defined between `ballots` and `vote`, the other functions should be defined between `_updatedRules` and `_roundtrip`

File: index.js [hunk 2]
```diff
@@ -275,6 +275,160 @@ Client.prototype.ballots = function (viewingId) {
...
+Client.prototype._getTransactionCSVRows = function (viewingIds) {
+    let txContribData = this._publisherVoteData(viewingIds)
+    var publishers = underscore.keys(txContribData)
+
+    var currency = txContribData[publishers[0]].contribution.currency
+    var headerRow = ['Publisher','Votes','Fraction','BTC', currency].join(',')
+
+    var rows = [headerRow]
+
+    rows = rows.concat(publishers.map(function (pub) {
```
Comment (mrose17) [inline index.js]: shouldn't there be some kind of CSV quoting for the publisher name?

File: index.js [hunk 3]
```diff
@@ -275,6 +275,160 @@ Client.prototype.ballots = function (viewingId) {
...
+
+    var rows = [headerRow]
+
+    rows = rows.concat(publishers.map(function (pub) { 
+        var pubRow = txContribData[pub]
+        return [pub,
+                pubRow.votes,
+                pubRow.fraction,
+                pubRow.contribution.satoshis / Math.pow(10, 10), 
+                pubRow.contribution.fiat.toFixed(2) + ' ' + pubRow.contribution.currency
```
Comment (mrose17) [inline index.js]: you should add a comment such as

        // TBD: update later to reflect correct number of fixed points for fiat currency in use

File: index.js (hunk 1 above)
Comment (diracdeltas) [inline index.js]: it's `_getTransactionCSVText`, which shouldn't begin with an underscore

**Pull request merged**